source/ns_cmdline.c

```diff
@@ -263,7 +272,6 @@ void cmd_init(cmd_print_t *outf)
         ns_list_init(&cmd.cmd_buffer);
         cmd.init = true;
     }
-    mbed_trace_exclude_filters_set(TRACE_GROUP);
```

Comment:
by the way, how it was designed to select which trace library is used (mbed-trace or unified logging) ?

Reply:
@jupe 
> when traces are enbled

Even when traces are enabled for other components of library, I don't see a way to enable them for `cmdL` trace group. So it should not affect.

> Propose to create macro which is ”null” or original function depend on if cli traces are enabled and mbed_trace is in use -> would not be breaking change.

Do you mean adding macro inside `#ifndef MBED_CLIENT_CLI_TRACE_ENABLE` like others?
```
+#undef mbed_trace_exclude_filters_set
+#define mbed_trace_exclude_filters_set(...)
```

Reply:
> by the way, how it was designed to select which trace library is used (mbed-trace or unified logging) ?

Enable flag `MBED_CONF_EXTERNAL_MBED_TRACE_ENABLED` to switch to mbed trace library, default will be unified tracing in mbed-os.

Reply:
Changing mbed-trace with unified logging by default is very high risk to break something. I recommend that unified logging would not be default at beginning so that there is time for teams to verify it work as expected in their use cases before it becomes default. 

another concern is that how unified logging and cli would work together from line print point of view. Is there similar “print mutexes” available which can be shares between unified logging and cli? Also queue usage for unified logging probably causes that traces and cli prints are not synchronized anymore which would causes confusion when tester reads test logs - e.g. command response is coming earlier than traces which belong to that command. Is there some solution for this?

Reply:
In order to disable traces, we have build time option. And for cli, option I suggested will have print enabled always (except in case of NDEBUG while everything is disabled)

I can add a commit to this same PR, and we can then verify that

Reply:
>  cli prints would contains then also trace formats?

No

Reply:
What is the reasoning of changing trace-logger? Is the trace-logger platform-specific or can it be used also on other platforms?

Reply:
Unified tracing is to support prints from all modules and libraries, it is for mbed-os platfrom.

Reply:
this is a bit problematic from client point of view because client supports multiple OS and mbed-OS is just one of them. As long as mbed-trace is only one which support all of client used OS’s they probably wont use anything else, unless there is very good reason for it.

Reply:
>  As long as mbed-trace is only one which support all of client used OS’s they probably wont use anything else

Because of this reason we have provided mbed-trace library wrappers on top on unified logging. From client point of view it will be using mbed-trace but the backend for mbed-os will be unified logging.

Reply:
For me it sounds a bit confusion from client point of view; Client assumes that traces behaviour exactly same way in all OS, but now mbedOS would behaviour slightly different by default, eg test apps would need mbed OS specific initializing.. Maybe its not an problem - up to them.

Anyway, this PR would disable CLI traces by default so cannot activate anymore in runtime by default. Probably it is not used so much that it is accepted -> Current changes are ok for me.